### PR TITLE
Fixed a bug in SourceState.createExtract

### PR DESCRIPTION
**Pull request by liyinan926**
Signed-off-by: Yinan Li <liyinan926@gmail.com>

### CODE REVIEW

**Comment by liyinan926** [comment]
@zliu41 can you review?

File: gobblin-api/src/main/java/gobblin/configuration/SourceState.java [hunk 1]
```diff
@@ -92,8 +93,12 @@ public SourceState(State properties, List<WorkUnitState> previousTaskStates) {
   public synchronized Extract createExtract(Extract.TableType type, String namespace, String table) {
     Extract extract = new Extract(this, type, namespace, table);
     while (extractSet.contains(extract)) {
```
Comment (zliu41) [inline gobblin-api/src/main/java/gobblin/configuration/SourceState.java]: Seems `extractSet` is not used anywhere. Why is this needed?
Also this method will be deprecated in #214 since properties in `SourceState` do not need to be added to `Extract`.

Comment (liyinan926) [inline gobblin-api/src/main/java/gobblin/configuration/SourceState.java]: `extractSet` is used here to make sure that each `Extract` has a unique `id`. I need this to unblock the Lynda gobblin job testing.

**Comment by zliu41** [comment]
LGTM
